org.mixedrealitytoolkit.input/Readers/PinchInputReader.cs

```diff
@@ -0,0 +1,441 @@
+// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clause
+
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.XR;
+using UnityEngine.XR.Interaction.Toolkit;
+using UnityEngine.XR.Interaction.Toolkit.Inputs;
+using UnityEngine.XR.Interaction.Toolkit.Inputs.Readers;
+
+namespace MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// A class that reads pinch selection input and values from <see cref="InputActionProperty"/>. If no action is set or if an action is not bound to a control,
+    /// the selection state will be driven by the <see cref="MixedReality.Toolkit.Subsystems.IHandsAggregatorSubsystem"/> subsystem's pinch amount.
+    /// </summary>
+    /// <remarks>
+    /// When using this class, ensure that the <see cref="MixedReality.Toolkit.Subsystems.IHandsAggregatorSubsystem"/> is available and enabled.
+    /// This is a workaround for device's without interaction profiles for hands. Once universal hand interaction profiles are available,
+    /// this class will be removed.
+    /// </remarks>
+    [DefaultExecutionOrder(XRInteractionUpdateOrder.k_XRInputDeviceButtonReader)]
+    public class PinchInputReader : MonoBehaviour, IXRInputButtonReader
+    {
+        /// <summary>
+        /// The state of the pinch input reader when <see cref="InputActionProperty"/> is not set or not bound to a control.
+        /// </summary>
+        private struct FallbackState
```

Comment:
Should/could we use Unity's `InteractionState` struct instead of defining our own? It also comes with helpers, like `SetFrameState`, which sets all four values in one method call.

Reply:
I don't think so.  XRI3 has lightly deprecated `InteractionState` ... and now states the following:

```cs
    /// <remarks>
    /// Use of this class should be avoided outside of the context of playback and recording.
    /// Instead, use <see cref="XRBaseInputInteractor.LogicalInputState"/> when possible.
    /// </remarks>
```

Reply:
And `LogicalInputState` looks like too much for this case.

Sort of related to this comment....Unity recommended implementing `IXRInputButtonReader` to handle this "No Hand Interaction Profile".  The pattern here is following the pattern in the XRI3 hand input sample scene.

Reply:
Same question then, but "Should/could we use Unity's _LogicalInputState_ struct instead of defining our own?" 😉

Reply:
> And LogicalInputState looks like too much for this case.

Beat me to my follow-up haha!

Reply:
@keveleigh ... So are you good with this? :)

Reply:
Ah yeah, that's fine!